src/shared/fdset.c

```diff
@@ -192,24 +192,39 @@ int fdset_new_listen_fds(FDSet **ret, bool unset) {
         return 0;
 }
 
-int fdset_close_others(FDSet *fds) {
-        _cleanup_free_ int *a = NULL;
-        size_t j = 0, m;
+int fdset_get_array(FDSet *fds, int **ret_array) {
+        unsigned j = 0, m;
         void *e;
+        int *a;
+
+        assert(ret_array);
 
         m = fdset_size(fds);
-        if (m > 0) {
-                a = new(int, m);
-                if (!a)
-                        return -ENOMEM;
+        if (m > INT_MAX) /* We want to be able to return an "int" */
```

Comment:
shouldn't this check if it's zero too?

Reply:
why?

this is just a paranoid check to ensure that `m` (which is unsigned) will fit into an int too, since that's how we'd ultimately like to return it.

Reply:
The previous code did - it allocated `a` only if `m > 0`. What happens if `m` is 0 here? You get a 1-sized array, that is actually empty? Isn't that a bit weird?